app-benchmarks/bonnie++/patches/bonnie++-1.98.patchset

```diff
@@ -0,0 +1,154 @@
+From 50ad9f4f492cc5645d5f9bdf099a4bf291244a8f Mon Sep 17 00:00:00 2001
+From: Kyle Ambroff-Kao <kyle@ambroffkao.com>
+Date: Sun, 8 Sep 2019 17:22:35 -0700
+Subject: [PATCH] Fix support for large files on Haiku
+
+---
+ bonnie++.cpp | 7 ++++---
+ port.h       | 9 +++++++--
+ port.h.in    | 7 ++++++-
+ zcav.cpp     | 6 +++---
+ zcav_io.cpp  | 4 ++--
+ 5 files changed, 22 insertions(+), 11 deletions(-)
+
+diff --git a/bonnie++.cpp b/bonnie++.cpp
+index eba83ef..d78c213 100644
+--- a/bonnie++.cpp
++++ b/bonnie++.cpp
+@@ -298,7 +298,7 @@ int main(int argc, char *argv[])
+       {
+         char *sbuf = _strdup(optarg);
+         char *size = strtok(sbuf, ":");
+-#ifdef _LARGEFILE64_SOURCE
++#if defined(_LARGEFILE64_SOURCE) || defined(__HAIKU__)
```

Comment:
Why not define _LARGEFILE64_SOURCE for Haiku instead?

Reply:
It could be done that way, although that would require some slightly more invasive changes since in this code `_LARGEFILE64_SOURCE` is a signal to use `lseek64`, `open64` and `creat64`.

```
#ifdef _LARGEFILE64_SOURCE
#define OFF_TYPE off64_t
#define file_lseek lseek64
#define file_creat creat64
#define file_open open64
...
```

I was going for a more minimal change but I'm happy to redo it if you think it's worth it.

Reply:
Wouldn't the diff be smaller if you instead keep the `_LARGEFILE64_SOURCE` define and change this:

```c
#ifdef _LARGEFILE64_SOURCE
#define OFF_TYPE off64_t
#define file_lseek lseek64
#define file_creat creat64
#define file_open open64
#else
#define OFF_TYPE off_t
#define file_lseek lseek
#define file_creat creat
#define file_open open
#endif
```

to

```c
#if defined(_LARGEFILE64_SOURCE) && !defined(__HAIKU__)
#define OFF_TYPE off64_t
#define file_lseek lseek64
#define file_creat creat64
#define file_open open64
#else
#define OFF_TYPE off_t
#define file_lseek lseek
#define file_creat creat
#define file_open open
#endif
```
?

Reply:
Yup, you're right. I just redid it that way by fixing the check for _LARGEFILE64_SOURCE in configure.in. I updated the recipe to depend on autoreconf and aclocal.